### DISCUSSION TITLE
How can i setup local access only for vaultwarden with traefik?

### DISCUSSION THREAD

**Discussion by rama31244** [dani-garcia/vaultwarden#4513] [Q&A]
I've got vaultwarden (running on docker on my unraid server) currently exposed to the internet through a cloudflare tunnel pointing to traefik reverse proxy but am looking to make it only available on my internal network (and on tailscale VPN) for increased security. I know that vaultwarden needs HTTPS to function but can't work out a way to make my subdomain (vaultwarden.mydomain.com) point to traefik so that it can generate a wildcard SSL certificate for vaultwarden. I tried to follow this guide linked below but when I set up the DNS record on cloudflare to point to my internal server address, it just leads to the unraid login page rather than vaultwarden and I don't have the ability to add a port number on the DNS record. How can I make the DNS record point to the traefik port of my unraid server?

[https://github.com/dani-garcia/vaultwarden/wiki/Running-a-private-vaultwarden-instance-with-Let's-Encrypt-certs](https://github.com/dani-garcia/vaultwarden/wiki/Running-a-private-vaultwarden-instance-with-Let%27s-Encrypt-certs)

**Comment by stefan0xC** [comment]
> when I set up the DNS record on cloudflare to point to my internal server address, it just leads to the unraid login page rather than vaultwarden

How have you configured traefik? On what port does traefik listen on? (If you don't want to specify a port you'd have to put traefik in charge of port `443` otherwise you'd also have to tell your browser about the port traefik listens on...)

> How can I make the DNS record point to the traefik port of my unraid server?

You can't and not sure if it makes sense conceptually. See https://www.cloudflare.com/learning/dns/what-is-dns/

In [the guide](https://github.com/dani-garcia/vaultwarden/wiki/Running-a-private-vaultwarden-instance-with-Let%27s-Encrypt-certs#getting-a-custom-caddy-build) you've linked [caddy-dns/cloudflare](https://github.com/caddy-dns/cloudflare) is used for configuring a ACME [DNS challenge](https://datatracker.ietf.org/doc/html/rfc8555#section-8.4).
With `traefik` you'd need to use the respective provider to let it automatically setup the challenge response using a TXT record (which is used to prove ownership of the domain). Cf. https://doc.traefik.io/traefik/https/acme/#dnschallenge
